### PR TITLE
Chri7325/linux llvm9

### PR DESCRIPTION
**Pull request by chri7325**
@dkalowsk The old runtimecore branch has been tagged as release/100.6.0 and I've changed it to now point to the final release of llvmorg-9.0.0. This branch ports all of the previous changes we made while also importing new projects and incorporating them into the build.

### CODE REVIEW

**Comment by chri7325** [comment]
Is it okay to merge this?

File: clang-tools-extra/constexpr-everything/main.cpp [hunk 1]
```diff
@@ -229,7 +229,7 @@ class ConstexprFunctionASTVisitor
 
     // Mark function as constexpr, the next ast visitor will use this
     // information to find constexpr vardecls
-    func->setConstexpr(true);
+    func->setConstexprKind(CSK_constexpr);
```
Comment (dkalowsk) [inline clang-tools-extra/constexpr-everything/main.cpp:232]: why is this changed needed from the mainline?

File: clang-tools-extra/templight/templight_driver.cpp [hunk 2]
```diff
@@ -365,7 +365,7 @@ static int ExecuteTemplightInvocation(CompilerInstance *Clang) {
   // This should happen AFTER plugins have been loaded!
   if (!Clang->getFrontendOpts().LLVMArgs.empty()) {
     unsigned NumArgs = Clang->getFrontendOpts().LLVMArgs.size();
-    auto Args = std::make_unique<const char *[]>(NumArgs + 2);
+    auto Args = make_unique<const char *[]>(NumArgs + 2);
```
Comment (dkalowsk) [inline clang-tools-extra/templight/templight_driver.cpp:368]: why is this change being made from the mainline import?

File: README.md [hunk 3]
```diff
@@ -1,5 +1,192 @@
...
 
-This directory and its subdirectories contain source code for LLVM,
-a toolkit for the construction of highly optimized compilers,
-optimizers, and runtime environments.
+This directory and its subdirectories contain source code for LLVM, a toolkit for the construction of highly optimized
+compilers, optimizers, and runtime environments.
+
+## RTC instructions
+
+RTC uses a few llvm tools that require custom fixes, custom checks are are not officially part of the llvm project quite
```
Comment (dkalowsk) [inline README.md]: `that require custom fixes and checks that are not... `?

double are are already on the line.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,192 @@
...
+  -DLIBCXXABI_ENABLE_STATIC_UNWINDER=ON \
+  -DLIBCXXABI_USE_LLVM_UNWINDER=ON \
+  \
+  -DLIBUNWIND_ENABLE_SHARED=OFF \
+  -DLIBUNWIND_USE_COMPILER_RT=ON \
+  -DLIBUNWIND_HERMETIC_STATIC_LIBRARY=ON
+
+# fix build.make in libcxx to have merge_archives.py command to include the full path to libc++abi
+sed -i "s%libc++abi.a%${HOME}/llvm-project/build_libcxx/lib/libc++abi.a%" projects/libcxx/src/CMakeFiles/cxx_static.dir/build.make
+make -j14 -k
```
Comment (dkalowsk) [inline README.md]: j14?  Weren't you just at j10 a few lines ago? :-)

**Review by dkalowsk — CHANGES_REQUESTED** [review]
One minor change needed in the readme, the rest looks good.  I did have some open questions to understand why the changes were made.

File: README.md (hunk 3 above)
Comment (chri7325) [reply README.md]: Nice catch

File: clang-tools-extra/constexpr-everything/main.cpp (hunk 1 above)
Comment (chri7325) [reply clang-tools-extra/constexpr-everything/main.cpp:232]: The mainline is aimed at clang 8 so it required this api change.

File: clang-tools-extra/templight/templight_driver.cpp (hunk 2 above)
Comment (chri7325) [reply clang-tools-extra/templight/templight_driver.cpp:368]: The mainline is set to clang 8. For clang 9, they're now using c++11 so there's no longer a need for their old custom make_unique calls.

**Review by dkalowsk — APPROVED** [review]
Thanks for the answers.
